apollo-router-core/src/layers/subgraph_cache.rs

```diff
@@ -0,0 +1,85 @@
+use crate::fetch::OperationKind;
+use crate::layer::ConfigurableLayer;
+use crate::layers::cache::{CachingLayer, CachingService, Sentinel};
+use crate::register_layer;
+use crate::{http_compat, Request, Response, SubgraphRequest, SubgraphResponse};
+use moka::sync::{Cache, CacheBuilder};
+use schemars::JsonSchema;
+use serde::Deserialize;
+use tower::{BoxError, Layer, Service};
+register_layer!("", "cache", SubgraphCachingLayer);
+
+#[derive(Deserialize, JsonSchema)]
+pub struct Config {
+    max_capacity: Option<usize>,
+    time_to_live: Option<String>,
+    time_to_idle: Option<String>,
```

Comment:
I'm slightly concerned about expressing these parameters as externally configurable items so early in the process. I'm not sure that we know enough about cache performance to be sure that we want such parameters to be externally configurable right now. We should at least discuss this because they are easy to add later, but probably more problematic to remove.

Reply:
Addressed by moving the layer to experimental.